packages/renderer/src/lib/deployments/deployment-utils.ts

```diff
@@ -29,16 +29,47 @@ export class DeploymentUtils {
     return humanizeDuration(uptimeInMs, { round: true, largest: 1 });
   }
 
+  refreshAge(deployment: DeploymentUI): string {
+    if (!deployment.created) {
+      return '';
+    }
+    // make it human friendly
+    return this.humanizeAge(deployment.created.toString());
+  }
+
   getDeploymentUI(deployment: V1Deployment): DeploymentUI {
+    const conditions = [];
+    if (deployment.status?.conditions) {
+      for (const con of deployment.status.conditions) {
+        const c: DeploymentCondition = {
+          type: con.type,
+          message: con.message,
+        };
+        conditions.push(c);
+      }
+    }
+    let status = 'STOPPED';
+    if (deployment.status?.readyReplicas && deployment.status?.readyReplicas > 0) {
+      if (deployment.status?.replicas === deployment.status?.readyReplicas) {
+        status = 'RUNNING';
+      } else {
+        status = 'DEGRADED';
+      }
+    }
     return {
       name: deployment.metadata?.name || '',
+      status: status,
       namespace: deployment.metadata?.namespace || '',
-      age: this.humanizeAge((deployment.metadata?.creationTimestamp || '').toString()),
+      created: deployment.metadata?.creationTimestamp,
+      age: deployment.metadata?.creationTimestamp
+        ? this.humanizeAge(deployment.metadata.creationTimestamp.toString())
```

Comment:
can use maybe the same approach then previous lines with ?? or || if there is no value
or we let humanizeAge accept undefined and it returns '' in that case

Reply:
I tried a couple ways and went with humanizeAge() accepting undefined since it seemed like the shortest/simplest code to me.